src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonObject.cs

```diff
@@ -70,6 +70,41 @@ internal JsonObject(JsonElement element, JsonNodeOptions? options = null) : this
             _jsonElement = element;
         }
 
+        internal override JsonNode InternalDeepClone()
+        {
+            if (_jsonElement.HasValue)
+            {
+                return new JsonObject(_jsonElement!.Value.Clone(), Options);
+            }
+
+            var jObject = new JsonObject(Options);
```

Comment:
Similarly, could the internal dictionary be presized given that we known the number of properties ahead of time?